<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>681</width>
    <height>638</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="groupBox_start">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>40</y>
      <width>171</width>
      <height>171</height>
     </rect>
    </property>
    <property name="title">
     <string>Start</string>
    </property>
    <widget class="QLineEdit" name="Start_0_0">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>40</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="Start_0_1">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>40</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="Start_0_2">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>40</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="Start_1_0">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>80</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="Start_1_1">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>80</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="Start_1_2">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>80</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="Start_2_0">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>120</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="Start_2_1">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>120</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="Start_2_2">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>120</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_end">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>40</y>
      <width>171</width>
      <height>171</height>
     </rect>
    </property>
    <property name="title">
     <string>End</string>
    </property>
    <widget class="QLineEdit" name="End_0_0">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>40</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="End_0_2">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>40</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="End_2_1">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>120</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="End_0_1">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>40</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="End_1_0">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>80</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="End_2_2">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>120</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="End_1_2">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>80</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="End_1_1">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>80</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="End_2_0">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>120</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QCheckBox" name="checkBox">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>210</y>
      <width>211</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>the first kind of f(n)</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="checkBox_2">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>210</y>
      <width>211</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>the second kind of f(n)</string>
    </property>
   </widget>
   <widget class="QPushButton" name="go_button">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>310</y>
      <width>93</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>&amp;Go</string>
    </property>
   </widget>
   <widget class="QPushButton" name="show_button">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>310</y>
      <width>121</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>&amp;Show Detail</string>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_process">
    <property name="geometry">
     <rect>
      <x>460</x>
      <y>40</y>
      <width>171</width>
      <height>171</height>
     </rect>
    </property>
    <property name="title">
     <string>Process</string>
    </property>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>40</y>
       <width>111</width>
       <height>111</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="1">
       <widget class="QLabel" name="process_0_1">
        <property name="text">
         <string>*</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="process_0_0">
        <property name="text">
         <string>*</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="process_1_0">
        <property name="text">
         <string>*</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="process_2_0">
        <property name="text">
         <string>*</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="process_1_1">
        <property name="text">
         <string>*</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QLabel" name="process_1_2">
        <property name="text">
         <string>*</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QLabel" name="process_0_2">
        <property name="text">
         <string>*</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLabel" name="process_2_1">
        <property name="text">
         <string>*</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QLabel" name="process_2_2">
        <property name="text">
         <string>*</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="Detail">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>340</y>
      <width>621</width>
      <height>231</height>
     </rect>
    </property>
    <property name="title">
     <string>Detail</string>
    </property>
    <widget class="QTextBrowser" name="textBrowser">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>30</y>
       <width>291</width>
       <height>191</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_1">
     <property name="geometry">
      <rect>
       <x>370</x>
       <y>90</y>
       <width>111</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Total Time:</string>
     </property>
    </widget>
    <widget class="QLabel" name="output_time_label">
     <property name="geometry">
      <rect>
       <x>470</x>
       <y>90</y>
       <width>72</width>
       <height>15</height>
      </rect>
     </property>
     <property name="text">
      <string>--ms</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>370</x>
       <y>130</y>
       <width>101</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Total Step:</string>
     </property>
    </widget>
    <widget class="QLabel" name="step_number_label">
     <property name="geometry">
      <rect>
       <x>470</x>
       <y>130</y>
       <width>72</width>
       <height>15</height>
      </rect>
     </property>
     <property name="text">
      <string>--</string>
     </property>
    </widget>
    <widget class="QPushButton" name="Full_tree_button">
     <property name="geometry">
      <rect>
       <x>490</x>
       <y>180</y>
       <width>89</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>Full Tree</string>
     </property>
    </widget>
   </widget>
   <widget class="QLabel" name="picture">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>370</y>
      <width>641</width>
      <height>201</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">image: url(:/new/prefix1/tongji.jpg);
</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QCheckBox" name="checkBox_3">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>250</y>
      <width>211</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>the third kind of f(n)</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="checkBox_4">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>250</y>
      <width>211</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>the forth kind of f(n)</string>
    </property>
   </widget>
   <zorder>picture</zorder>
   <zorder>groupBox_start</zorder>
   <zorder>groupBox_end</zorder>
   <zorder>checkBox</zorder>
   <zorder>checkBox_2</zorder>
   <zorder>go_button</zorder>
   <zorder>show_button</zorder>
   <zorder>groupBox_process</zorder>
   <zorder>Detail</zorder>
   <zorder>checkBox_3</zorder>
   <zorder>checkBox_4</zorder>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>681</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_About">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="action_About_us"/>
    <addaction name="action_Usage"/>
   </widget>
   <addaction name="menu_About"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="action_About_us">
   <property name="text">
    <string>&amp;About us</string>
   </property>
  </action>
  <action name="action_Usage">
   <property name="text">
    <string>&amp;Usage</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>go_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>Go_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>478</x>
     <y>304</y>
    </hint>
    <hint type="destinationlabel">
     <x>424</x>
     <y>333</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>show_button</sender>
   <signal>toggled(bool)</signal>
   <receiver>Detail</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>572</x>
     <y>306</y>
    </hint>
    <hint type="destinationlabel">
     <x>576</x>
     <y>344</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>show_button</sender>
   <signal>toggled(bool)</signal>
   <receiver>picture</receiver>
   <slot>setHidden(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>630</x>
     <y>298</y>
    </hint>
    <hint type="destinationlabel">
     <x>663</x>
     <y>386</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Full_tree_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>Show_tree_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>562</x>
     <y>523</y>
    </hint>
    <hint type="destinationlabel">
     <x>656</x>
     <y>563</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>Go_clicked()</slot>
  <slot>Show_tree_clicked()</slot>
 </slots>
</ui>
